R/glue.R

```diff
@@ -287,12 +287,21 @@ trim <- function(x) {
 
 #' @export
 print.glue <- function(x, ..., sep = "\n") {
-  x[is.na(x)] <- style_na(x[is.na(x)])
-
-  if (length(x) > 0) {
-    cat(x, ..., sep = sep)
+  if (length(x) == 0) {
+    cat("<Length-0 glue vector>\n")
+    return(invisible(x))
   }
 
+  id <- format(paste0("[", seq_along(x), "] "), justify = "right")
+  indent <- paste0(style_grey(id, "|"), " ")
+  exdent <- paste0(strrep(" ", nchar(id[[1]])), style_grey("|"), " ")
+
+  out <- x
+  out[is.na(out)] <- style_red("NA")
+  out <- paste0(indent, out)
+  out <- gsub("\n", paste0("\n", exdent), out)
```

Comment:
I tried a number of things to break this. The only thing I can think of is that it can make colored values show up incorrectly

```r
# cli::col_red("foo\nbar")
x <- "\033[31mfoo\nbar\033[39m"
cat(x)

glue::glue(x)
```
<img width="326" alt="Screen Shot 2023-01-27 at 4 16 55 PM" src="https://user-images.githubusercontent.com/19150088/215202505-83d35fd5-709c-4a4c-b94d-efe372ed896f.png">

Reply:
Presumably we can fix this if we set our minds to it, although it would be much more pleasant to do so using cli, which is only in `Suggests`.